pkg/apis/serving/k8s_validation.go

```diff
@@ -133,8 +141,12 @@ func validateVolume(volume corev1.Volume) *apis.FieldError {
 			errs = errs.Also(validateProjectedVolumeSource(proj).ViaFieldIndex("projected", i))
 		}
 	}
+	if vs.EmptyDir != nil {
+		specified = append(specified, "emptyDir")
+		errs = errs.Also(validateEmptyDirFields(vs.EmptyDir).ViaField("emptyDir"))
+	}
 	if len(specified) == 0 {
-		errs = errs.Also(apis.ErrMissingOneOf("secret", "configMap", "projected"))
+		errs = errs.Also(apis.ErrMissingOneOf("secret", "configMap", "projected", "emptyDir"))
```

Comment:
presumably "emptydir" should only be in this list if the flag is enabled?

Reply:
correct